main.rb

```diff
@@ -0,0 +1,151 @@
+# rubocop:disable Metrics/CyclomaticComplexity
+# rubocop:disable Metrics/PerceivedComplexity
+module Enumerable
+  def my_each
+    return enum_for unless block_given?
+
+    array = is_a?(Range) ? to_a : self
+    array.length.times { |index| yield(array[index]) }
+    array
+  end
```

Comment:
# my_each
1. [ ] When a block is given when self is a range returns the range itself after calling the given block once for each element in self
### Test 👉 `expect(range.my_each(&block)).to eq(range.each(&block))`
|Expected|Got|
|:---:|:---:|
|` 5..50`|`[5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50]`|
##
2. [ ] When a block is given when self is a hash behaves like #each calls the given block once for each element in self.
### Test 👉 `expect(my_each_output).to eq(each_output)`
|Expected|Got|
|:---:|:---:|
|`"[:a, 1][:b, 2][:c, 3][:d, 4][:e, 5]"`|`""`|